libraries/chain/transaction_context.cpp

```diff
@@ -9,26 +9,25 @@
 namespace eosio { namespace chain {
 
    transaction_context::transaction_context( controller& c,
-                                             transaction_trace_ptr& trace_ptr,
                                              const signed_transaction& t,
-                                             const transaction_id_type& trx_id,
-                                             fc::time_point d,
-                                             fc::time_point p,
-                                             uint64_t initial_net_usage,
-                                             uint64_t initial_cpu_usage         )
+                                             const transaction_id_type& trx_id )
    :control(c)
    ,trx(t)
    ,id(trx_id)
    ,undo_session(c.db().start_undo_session(true))
    ,trace(std::make_shared<transaction_trace>())
-   ,deadline(d)
-   ,published(p)
    ,net_usage(trace->net_usage)
    ,cpu_usage(trace->cpu_usage)
    {
       trace->id = id;
-      trace_ptr = trace;
       executed.reserve( trx.total_actions() );
+      is_input = false;
+      apply_context_free = false;
```

Comment:
This line should be removed. You are setting apply_context_free to false here and nowhere does it become true.